var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -232,13 +238,38 @@ def nvariant_bool(feature):
                 '-DPARAVIEW_QT_VERSION=%s' % spec['qt'].version[0],
             ])
 
+        # CMake flags for python are different for different versions
+        # Make sure Spack uses the right cmake flags
         if '+python' in spec or '+python3' in spec:
-            cmake_args.extend([
-                '-DPARAVIEW_ENABLE_PYTHON:BOOL=ON',
-                '-DPYTHON_EXECUTABLE:FILEPATH=%s' %
-                spec['python'].command.path,
-                '-DVTK_USE_SYSTEM_MPI4PY:BOOL=%s' % variant_bool('+mpi')
-            ])
+            if spec.satisfies('@5.8:'):
+                cmake_args.append('-DPARAVIEW_USE_PYTHON:BOOL=ON')
+                if '+python3' in spec:
+                    cmake_args.append('-DPARAVIEW_PYTHON_VERSION:STRING=3')
+                else:
+                    cmake_args.append('-DPARAVIEW_PYTHON_VERSION:STRING=2')
+            elif spec.satisfies('@5.7:'):
+                cmake_args.extend([
+                    '-DPARAVIEW_ENABLE_PYTHON:BOOL=ON',
+                    '-DPYTHON_EXECUTABLE:FILEPATH=%s' %
+                    spec['python'].command.path
+                ])
+                if '+python3' in spec:
+                    cmake_args.append('-DPARAVIEW_PYTHON_VERSION:STRING=3')
+                else:
+                    cmake_args.append('-DPARAVIEW_PYTHON_VERSION:STRING=2')
+            else:
+                cmake_args.extend([
+                    '-DPARAVIEW_ENABLE_PYTHON:BOOL=ON',
+                    '-DPYTHON_EXECUTABLE:FILEPATH=%s' %
+                    spec['python'].command.path
+                ])
+                if '+python3' in spec:
+                    cmake_args.append('-DVTK_PYTHON_VERSION:STRING=3')
+                else:
+                    cmake_args.append('-DVTK_PYTHON_VERSION:STRING=2')
+            cmake_args.append('-DVTK_USE_SYSTEM_MPI4PY:BOOL=%s' %
+                              variant_bool('+mpi'))
+
```

Comment:
I believe this entire logic block can be shrunk and simplified for readability with:
```
        if '+python' in spec or '+python3' in spec:
            py_use_opt = 'USE' if spec.satisfies('@5.8:') else 'ENABLE'
            py_ver_opt = 'PARVIEW' if spec.satisfies('@5.7:') else 'VTK'
            py_ver_val = 3 if 'python3' in spec else 2
            cmake_args.extend([
                '-DPARAVIEW_%s_PYTHON:BOOL=ON' % py_use_opt,
                '-DPYTHON_EXECUTABLE:FILEPATH=%s' % spec['python'].command.path,
                '-DVTK_USE_SYSTEM_MPI4PY:BOOL=%s' % variant_bool('+mpi'),
                '-D%s_PYTHON_VRERSION:STRING=%d' % (py_ver_opt, py_ver_val)
            ])
```

Reply:
Changed the logic to this. One thing to note is ParaView 5.8.0 ignores -DPYTHON_EXECUTABLE, but I don't think that is an issue.